tests/queries/0_stateless/03246_range_literal_replacement_works.sql

```diff
@@ -1,13 +1,5 @@
-CREATE TABLE my_table (
-  str String,
-) ORDER BY str;
+CREATE TABLE 03246_range_literal_replacement_works (id UInt8) Engine=Memory;
 
-INSERT INTO my_table VALUES
-(
-CASE WHEN
-  (0 BETWEEN 0 AND 2) THEN 'ok' ELSE
-  'wat'
-END
-);
+INSERT INTO 03246_range_literal_replacement_works VALUES (1 BETWEEN 0 AND 2);
```

Comment:
Why do we need to change 0 -> 1 here?

Reply:
A `1` to show success in the expected to me looks better in the expected file than a `0`. Anyway, the important part is that there's a range. This comes from https://github.com/ClickHouse/ClickHouse/pull/70122